lib/cocoapods/installer/xcode/single_pods_project_generator.rb

```diff
@@ -0,0 +1,32 @@
+module Pod
+  class Installer
+    class Xcode
+      # The {SinglePodsProjectGenerator} handles generation of the 'Pods/Pods.xcodeproj'
+      #
+      class SinglePodsProjectGenerator < PodsProjectGenerator
+        # Generates single `Pods/Pods.xcodeproj`.
+        #
+        # @return [PodsProjectGeneratorResult]
+        #
+        def generate!
+          project_path = sandbox.project_path
+          build_configurations = analysis_result.all_user_build_configurations
+          platforms = aggregate_targets.map(&:platform)
+          object_version = aggregate_targets.map(&:user_project).compact.map { |p| p.object_version.to_i }.min
+          project_generator = ProjectGenerator.new(sandbox, project_path, pod_targets, build_configurations,
+                                                   platforms, object_version, config.podfile_path)
+          project = project_generator.generate!
+          sandbox.project = project
```

Comment:
this remains very ugly. The fact we are now doubling it I recommend we fix it before we land this.

Reply:
https://github.com/CocoaPods/CocoaPods/pull/8259